### PR TITLE
#901 ListNotNulls immutable and null iterators

### PR DESCRIPTION
**Pull request by DanielPalafox**

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#934](https://github.com/yegor256/cactoos/issues/934) is now in scope, role is `REV`

File: src/main/java/org/cactoos/list/ListNoNulls.java [hunk 1]
```diff
@@ -72,7 +73,15 @@ public boolean contains(final Object item) {
 
     @Override
     public Iterator<T> iterator() {
-        return new CollectionNoNulls<>(this.list).iterator();
+        final Iterator<T> iterator;
+        if (this.list == null) {
+            iterator = Collections.emptyIterator();
+        } else {
+            iterator = new CollectionNoNulls<>(
+            Collections.unmodifiableList(this.list)
```
Comment (llorllale) [inline src/main/java/org/cactoos/list/ListNoNulls.java]: @DanielPalafox can you indent this line? I'm surprised it passed qulice checks.

File: src/main/java/org/cactoos/list/ListNoNulls.java [hunk 2]
```diff
@@ -72,7 +73,15 @@ public boolean contains(final Object item) {
 
     @Override
     public Iterator<T> iterator() {
-        return new CollectionNoNulls<>(this.list).iterator();
+        final Iterator<T> iterator;
+        if (this.list == null) {
```
Comment (llorllale) [inline src/main/java/org/cactoos/list/ListNoNulls.java]: @DanielPalafox I'm not sure what this null-check is for

File: src/test/java/org/cactoos/list/ListNoNullsTest.java [hunk 3]
```diff
@@ -101,4 +101,34 @@ public void removeThrowsErrorIfValueNull() {
...
+        new ListNoNulls<>(
+            list
+        ).removeIf(value -> value == 1);
+    }
+
+    @Test
+    public void iteratorThrowsErrorWithNullValues() {
+        this.exception.expect(IllegalStateException.class);
+        final ArrayList<Integer> list = new ArrayList<>(2);
+        list.add(1);
```
Comment (llorllale) [inline src/test/java/org/cactoos/list/ListNoNullsTest.java]: @DanielPalafox why add this element if the test is for null values?

File: src/test/java/org/cactoos/list/ListNoNullsTest.java [hunk 4]
```diff
@@ -101,4 +101,34 @@ public void removeThrowsErrorIfValueNull() {
...
     }
+
+    @Test
+    public void iteratorThrowsErrorWithRemoveOperation() {
+        this.exception.expect(UnsupportedOperationException.class);
+        final ArrayList<Integer> list = new ArrayList<>(1);
+        list.add(1);
+        new ListNoNulls<>(
+            list
+        ).removeIf(value -> value == 1);
```
Comment (llorllale) [inline src/test/java/org/cactoos/list/ListNoNullsTest.java]: @DanielPalafox the method intended for test `Iterator.remove`, not `List.removeIf`

**Comment by llorllale** [comment]
@DanielPalafox left you some comments

Also, please add a description to this PR

**Comment by llorllale** [comment]
@DanielPalafox ping

**Comment by llorllale** [comment]
@DanielPalafox ping

**Comment by llorllale** [comment]
@DanielPalafox closed due to inactivity

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#934` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#934](https://github.com/yegor256/cactoos/issues/934) is now out of scope
